frontend/src/modules/studySamples/services.ts

```diff
@@ -163,64 +129,61 @@ export async function buildStudySamplesFromWorkflow(
 	const usersByID = createItemsByID(users)
 
-	completedSamplesByStudy.forEach(stepHistory => {
-		const step = stepMap.get(stepHistory.step_order)
-		if (step) {
-			const processMeasurement = processMeasurementsByID[stepHistory.process_measurement]
-			const process = processMeasurement ? processesByID[processMeasurement.process] : undefined
-			const user = process && process.created_by ? usersByID[process.created_by] : stepHistory.created_by ? usersByID[stepHistory.created_by] : undefined
-			
-			const completedSample : CompletedStudySample = {
-				id: stepHistory.id,
-				sampleID: stepHistory.sample,
-				generatedSampleID: processMeasurement?.child_sample,
-				processID: processMeasurement?.process,
-				processMeasurementID: stepHistory.process_measurement,
-				executionDate: processMeasurement ? processMeasurement?.execution_date : timestampStringAsDate(stepHistory.created_at),
-				executedBy: user?.username,
-				comment: processMeasurement?.comment,
-				removedFromWorkflow: stepHistory.workflow_action === 'DEQUEUE_SAMPLE'
-			}
-			step.completed.push(completedSample)
-		}
+	const completedStudySamples = completedSamples.map((completedSample) => {
+		const processMeasurement = processMeasurementsByID[completedSample.process_measurement]
+		const process = processMeasurement ? processesByID[processMeasurement.process] : undefined
+		const user = process && process.created_by ? usersByID[process.created_by] : completedSample.created_by ? usersByID[completedSample.created_by] : undefined
+
+		return {
+			id: completedSample.id,
+			sampleID: completedSample.sample,
+			generatedSampleID: processMeasurement?.child_sample,
+			processID: processMeasurement?.process,
+			processMeasurementID: completedSample.process_measurement,
+			executionDate: processMeasurement ? processMeasurement?.execution_date : timestampStringAsDate(completedSample.created_at),
+			executedBy: user?.username,
+			comment: processMeasurement?.comment,
+		} as CompletedStudySample
 	})
 
-	// Return the steps in the step order
-	const steps = Array.from(stepMap.values()).sort((a, b) => a.stepOrder - b.stepOrder)
-
 	return {
-		steps
+		result: completedStudySamples,
+		count: totalCount
 	}
 }
 
-export async function fetchSamplesAtStepOrder(studyID: FMSId, stepOrderID: FMSId) {
-
+async function fetchSamplesAtStepOrder(studyID: FMSId, stepOrderID: number, offset = 0, limit = 10) {
 	const studySettingsByID = selectStudySettingsByID(store.getState())
 
 	// Get the current set of filters and sort order from UX settings for study and step
-	let options = {}
+	let options = { limit } as any
 	const settings = studySettingsByID[studyID]?.stepSettings[stepOrderID]
 	if (settings) {
 		const serializedFilters = settings.filters ? serializeFilterParamsWithDescriptions(settings.filters) : {}
 		const ordering = settings.sortBy ? serializeSortByParams(settings.sortBy) : undefined
-		options = {ordering, ...serializedFilters}
+		const limit = settings.pageSize
```

Comment:
This is weird here ... you get the page size as prop and also fetch it from the settings ... this is likely to cause issues (someone calls the function with a limit and expects to override the settings and instead get slapped back to the settings value ...). I would expect either the function to have limit as prop or fetch it from the settings not both. If both are used the settings.pagSize should be part of the default value (like the first default the second default being 10) and the default value of limit should be set to undefined in order to be able to tell if a value was passed down.